add.go

```diff
@@ -172,3 +192,36 @@ func (b *Builder) Add(destination string, extract bool, source ...string) error
 	}
 	return nil
 }
+
+// user returns the user (and group) information which the destination should belong to.
+func (b *Builder) user(mountPoint string, options AddAndCopyOptions) (specs.User, error) {
+	if options.Chown != "" {
+		return getUser(mountPoint, options.Chown)
+	}
+	return getUser(mountPoint, b.User())
+}
+
+// setOwner sets the uid and gid owners of a given path.
+// If path is a directory, recursively changes the owner.
+func setOwner(path string, user specs.User) error {
+	fi, err := os.Stat(path)
+	if err != nil {
+		return errors.Wrapf(err, "error reading %q", path)
+	}
+	if fi.IsDir() {
+		err2 := filepath.Walk(path, func(p string, info os.FileInfo, we error) error {
+			if err3 := os.Chown(p, int(user.UID), int(user.GID)); err3 != nil {
```

Comment:
`os.Chown()` follows symlinks, and since we're not doing any of this in a chrooted subprocess, we could open ourselves up to something like #66 again.  I think using `os.Lchown()` is a first step here.

Reply:
Hm, interesting perspective, thank you!